fedimint-server/src/consensus/mod.rs

```diff
@@ -270,14 +263,15 @@ impl FedimintConsensus {
                 .filter_conflicts(|(_, tx)| tx)
                 .partitioned();
 
-            let mut db_batch = DbBatch::new();
-            let mut batch_tx = db_batch.transaction();
+            let mut dbtx = self.db.begin_transaction();
+            dbtx.set_tx_savepoint();
```

Comment:
At minimum, shouldn't starting a transaction imply creating a savepoint?

Reply:
Good idea. Added in new commit.